local_planner/src/nodes/common.cpp

```diff
@@ -54,27 +54,42 @@ PolarPoint cartesianToPolar(double x, double y, double z,
 Eigen::Vector2i polarToHistogramIndex(const PolarPoint& p_pol, int res) {
   // TODO change logic here, as 0,0 are valid index
   Eigen::Vector2i ev2(0, 0);
-  float e = p_pol.e;
-  float z = p_pol.z;
-  if (res <= 0.f || e < -90.f || e > 90.f || z < -180.f || z > 180.f) {
-    return ev2;
+  PolarPoint p_wrapped = wrapPolar(p_pol);
+  // elevation angle to y-axis histogram index
+  if (p_wrapped.e == 90.0f) {
+    p_wrapped.e = 89.0f;
   }
-  if (e == 90.f) {
-    e = 89;
+  p_wrapped.e += 90.0f;
+  p_wrapped.e = p_wrapped.e + (res - (static_cast<int>(p_wrapped.e) % res));
+  ev2.y() = p_wrapped.e / res - 1;
```

Comment:
These three lines are cryptic and I don't really understand them. What's everyone's confidence in these? I would suggest something like the one-liner: `ev2.y() = floor(p_wrapped.e / res + 90.f / res);`
At the end of the day it's a floor of a linear function that maps elevation -90° to bin 0 and +90° to the highest bin N - 1 (= 180/res -1)

Reply:
agree, changed to the proposed solution